### ISSUE TITLE
options.mains does not resolve relative to src when using fn format of .neutrinorc

### ISSUE THREAD

**Issue by timkelty** [mozilla-neutrino/neutrino-dev#594]
e.g.
```
module.exports = neutrino => {
  Object.assign(neutrino.options, {
    output: 'public/build',
    mains: {
      index: 'js/index'
    },
  });
  neutrino.use('@neutrinojs/web';
};
```

**Comment by timkelty** [comment]
This can be avoided by using the object format and specifying `options`, which is probably smarter anyway.

If this is expected, there should probably be some sort of warning in the docs.

**Comment by timkelty** [comment]
I didn't know about `neutrino.mergeOptions` 👏 . That's perfect.

Perhaps we should throw an error or warning if you try to overwrite `neutrino.options` completely?

**Comment by eliperelman** [comment]
> Perhaps we should throw an error or warning if you try to overwrite neutrino.options completely?

This would be really hard to catch unless we had a proxy for the `neutrino` object, which would probably be more expensive than necessary.

> FWIW - I think we should also encourage folks to use the object format to avoid these kind of problems, even if they want to resort to fn for `use`.

Totally agree. In projects, we want users to be using object format when possible.

**Comment by timkelty** [comment]
> This would be really hard to catch unless we had a proxy for the neutrino object, which would probably be more expensive than necessary.

Perhaps just a note in the docs then, somewhere in here?
 https://neutrino.js.org/middleware/#function-format

**Comment by eliperelman** [comment]
Addressed in #614.
